File: Python/Easy/العناصر المتشابهة في المصفوفة.py

# Creator: OmarAfet
# https://profile.satr.codes/OmarAfet/public/overview
# https://github.com/OmarAfet

def get_duplicate_elements(arr: list[int]) -> list[int]:
    x = []
    y = []
    for i in arr:
        if i in x and i not in y:
            y.append(i)
            pass
        else:
            x.append(i)
    return y

File: JavaScript/Easy/العناصر المتشابهة في المصفوفة.js
/*

Creator: OmarAfet
https://profile.satr.codes/OmarAfet/public/overview
https://github.com/OmarAfet

*/

function get_duplicate_elements(arr) {
	let x = [];
	let y = [];

	for (let i = 0; i < arr.length; i++) {
		const e = arr[i];

		if (x.includes(e) && !y.includes(e)) {
			y.push(e);
		} else {
			x.push(e);
		}
	}

	return y;
}
